internal/painter/software/draw.go

```diff
@@ -82,10 +82,16 @@ func drawImage(c fyne.Canvas, img *canvas.Image, pos fyne.Position, base *image.
 }
 
 func drawPixels(x, y, width, height int, mode canvas.ImageScale, base *image.NRGBA, origImg image.Image, clip image.Rectangle) {
+	if origImg.Bounds().Dx() == width && origImg.Bounds().Dy() == height {
+		// do not scale or duplicate image since not needed, draw directly
+		drawTex(x, y, width, height, base, origImg, clip)
+		return
+	}
+
 	scaledBounds := image.Rect(0, 0, width, height)
 	scaledImg := image.NewNRGBA(scaledBounds)
 	switch mode {
-	case canvas.ImageScalePixels:
+	case canvas.ImageScalePixels, canvas.ImageScaleFastest:
```

Comment:
Thanks for changing this method, it does need to scale.
However I think the addition of Fastest here will give a pixelated output - was that your intent?

Reply:
That's true. I'm guessing `BiLinear` would be too slow, so maybe `ApproxBiLinear`?